File: spec/python/test_process_coerce_bytes.py
import unittest

from process_coerce_bytes import ProcessCoerceBytes

class TestProcessCoerceBytes(unittest.TestCase):
    def test_process_coerce_bytes(self):
        r = ProcessCoerceBytes.from_file("src/process_coerce_bytes.bin")

        self.assertEqual(r.records[0].flag, 0)
        self.assertEqual(r.records[0].buf, b"AAAA")
        self.assertEqual(r.records[1].flag, 1)
        self.assertEqual(r.records[1].buf, b"BBBB")

File: spec/lua/test_process_coerce_bytes.lua
local luaunit = require("luaunit")

require("process_coerce_bytes")

TestProcessCoerceBytes = {}

function TestProcessCoerceBytes:test_process_coerce_bytes()
    local r = ProcessCoerceBytes:from_file("src/process_coerce_bytes.bin")

    luaunit.assertEquals(r.records[1].flag, 0)
    luaunit.assertEquals(r.records[1].buf, "AAAA")
    luaunit.assertEquals(r.records[2].flag, 1)
    luaunit.assertEquals(r.records[2].buf, "BBBB")
end
